allwinner-hal/src/dma/register.rs

```diff
@@ -23,35 +23,357 @@ pub struct RegisterBlock {
+pub struct ChannelPauseRegister(pub u32);
+
+/// Channel Start Address Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelStartAddrRegister(pub u32);
+
+/// Channel Configuration Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelConfigRegister(pub u32);
+
+/// Channel Current Source Address Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelCurrentSrcAddrRegister(pub u32);
+
+/// Channel Current Destination Address Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelCurrentDestAddrRegister(pub u32);
+
+/// Channel Byte Counter Left Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelByteCounterLeftRegister(pub u32);
+
+/// Channel Parameter Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelParameterRegister(pub u32);
+
+/// Channel Mode Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelModeRegister(pub u32);
+
+/// Channel Former Descriptor Address Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelFormerDescAddrRegister(pub u32);
+
+/// Channel Package Number Register
+#[repr(transparent)]
+#[derive(Copy, Clone, Default)]
+pub struct ChannelPackageNumRegister(pub u32);
+
 /// DMAC channel registers.
 #[repr(C)]
 pub struct ChannelRegisterBlock {
     /// DMAC Channel Enable Register.
-    pub enable: RW<u32>,
+    pub enable: RW<ChannelEnableRegister>,
     /// DMAC Channel Pause Register.
-    pub pause: RW<u32>,
+    pub pause: RW<ChannelPauseRegister>,
     /// DMAC Channel Start Address Register.
-    pub start_addr: RW<u32>,
+    pub start_addr: RW<ChannelStartAddrRegister>,
     /// DMAC Channel Configuration Register.
-    pub config: RO<u32>,
+    pub config: RO<ChannelConfigRegister>,
     /// DMAC Channel Current Source Register.
-    pub current_src_addr: RO<u32>,
+    pub current_src_addr: RO<ChannelCurrentSrcAddrRegister>,
     /// DMAC Channel Current Destination Register.
-    pub current_destination: RO<u32>,
+    pub current_destination: RO<ChannelCurrentDestAddrRegister>,
     /// DMAC Channel Byte Counter Left Register.
-    pub byte_counter_left: RO<u32>,
+    pub byte_counter_left: RO<ChannelByteCounterLeftRegister>,
     /// DMAC Channel Parameter Register.
-    pub parameter: RO<u32>,
+    pub parameter: RO<ChannelParameterRegister>,
     _reserved0: [u8; 0x8],
     /// DMAC Mode Register.
-    pub mode: RW<u32>,
+    pub mode: RW<ChannelModeRegister>,
     /// DMAC Former Descriptor Address Register.
-    pub former_desc_addr: RO<u32>,
+    pub former_desc_addr: RO<ChannelFormerDescAddrRegister>,
     /// DMAC Package Number Register.
-    pub package_num: RO<u32>,
+    pub package_num: RO<ChannelPackageNumRegister>,
     _reserved1: [u8; 0xC],
 }
 
+impl ChannelEnableRegister {
+    // DMA Channel Enable (Bit 0)
+    pub const DMA_EN: u32 = 0x1 << 0;
```

Comment:
```suggestion
    const DMA_EN: u32 = 0x1 << 0;
```

Those constants not need to be public.